sources/packages/backend/apps/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts

```diff
@@ -382,11 +391,10 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
    * only records with coeStatus defined as 'Required' will be considered.
    * @returns first disbursement schedule, if any.
    */
-  async getFirstDisbursementSchedule(options: {
-    disbursementScheduleId?: number;
-    applicationId?: number;
-    onlyPendingCOE?: boolean;
-  }): Promise<DisbursementSchedule> {
+  async getFirstDisbursementScheduleByApplication(
```

Comment:
How do we ensure that the disbursement belongs to the application's current assessment?
I would say that we can start the query by the application and use the `application.currentAssessment` to return the `DisbursementSchedule`.

Reply:
Yes that makes sense and seems to be a great idea to get the disbursements which go through COE process.

Reply:
As we talked the COE status must come always from the application's current assessment/reassessment and the COE will be always confirmed by the institution no matter the "Original Assessment" or any reassessment, as per mentioned by @JasonCTang 
@ann-aot @guru-aot @andrepestana-aot FYI